packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/metadata.py

```diff
@@ -131,6 +131,8 @@ def _validate_default_value_type(parameter_metadata: ParameterMetadata) -> None:
             expected_element_type = type(
                 get_type_default(parameter_metadata.type, False, type(default_value[0]))
             )
+        else:
+            expected_element_type = expected_type
```

Comment:
This sets the expected element type to list, which doesn't make sense to me. When default_value is an empty list, that means we don't know what the element type is. It doesn't mean the element type is another list.

Please indent the call to _validate_default_value_type_for_repeated_type so it's part of the body of the if-statement:
```
        if len(default_value) > 0:
            expected_element_type = type(
                get_type_default(parameter_metadata.type, False, type(default_value[0]))
            )
            _validate_default_value_type_for_repeated_type(
                default_value,
                expected_type,
                expected_element_type,
                enum_values_annotation,
                display_name,
            )
```